### DISCUSSION TITLE
Check filters on struct data

### DISCUSSION THREAD

**Discussion by glitchedgitz** [pocketbase/pocketbase#2887] [Q&A] [answered]
Hi there,

Is there any way to check if the filters `created >= "2022-01-01 00:00:00" && someField1 != someField2` on struct data?
Something like this
```go

if checkFilters(dataRow, filter){
    // ...
}
```

**Comment by ganigeorgiev** [comment]
There is no such method and I'm not sure that I understand what do you refer by "struct data". Could you elaborate a little more on your use case and what you are trying to do?

**Reply by glitchedgitz** [reply]
Thanks for the fast response.

It's a golang application utilizing pocketbase SDK/API (not pocketbase framwork)

I want to check data aganist the filters before adding it to the collection. Otherwise, data goes to another collection.
Now instead of `if-else`, if I can check it with the filter string, It would be really great.


struct data means struct object, let say if we have this data, now checking these together
```go
var newData = UserData{
    Host :"google.com",
    Port :"8080",
    Path : "/search"
}

var filters = "created >= "2022-01-01 00:00:00" && someField1 != someField2"


```

**The hard solution**
One solution is to add data to a `temp` collection and fetch it with `id` and other filters.
Then I can check if the results are empty or not.

**Possible solution**
I was checking your code and for testing purposes you have created dummy db, so it might be possible to solve this with `pocketbase/dbx` in more efficient way.

**Reply by ganigeorgiev** [reply, marked as answer]
The current filters cannot be used for general expression evaluation. The filters parser generates an AST from which we create a SQL query and we don't have implementation for raw data because so far we don't have such use case in PocketBase.

There are several existing libraries that you can check for general expression evaluation:
- https://github.com/antonmedv/expr
- https://github.com/PaesslerAG/gval
- https://github.com/google/cel-spec
- and probably others...

In the upcoming v0.17.0 release there will be 2 new helpers [`Dao.FindRecordsByFilter()`](https://github.com/pocketbase/pocketbase/blob/7bb33d4453453731cd01d6d810105518cd405593/daos/record.go#L239) and [`Dao.FindFirstRecordByFilter`](https://github.com/pocketbase/pocketbase/blob/7bb33d4453453731cd01d6d810105518cd405593/daos/record.go#L303) but both operates on DB Record data.

**Reply by glitchedgitz** [reply]
Thanks @ganigeorgiev 
I checked and seems like [google/cel-spec](https://github.com/google/cel-spec) can match the pocketbase syntax with some configurations.
